packages/next/build/index.ts

```diff
@@ -423,23 +425,29 @@ export default async function build(dir: string, conf = null): Promise<void> {
             runtimeEnvConfig
           )
 
-          if (result.isHybridAmp) {
-            hybridAmpPages.add(page)
-          }
+          if (isErrorPage) {
+            if (result.static && customAppGetInitialProps === false) {
```

Comment:
This doesn't seem fully correct though, it matches only if there's a custom _app that has getInitialProps, what if _error also has getInitialProps? I think we need to pre-generate 404, 500 and 501 or something. Also when the user has no custom _error we can just always render to static 👍

Reply:
it matches if the custom _app doesn't have `getInitialProps` - if `_error.js` has a custom `getInitialProps` it doesn't prerender, that's because of the dynamic use cases that were added in the RFC